src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/attachment-unit-form/attachment-unit-form.component.ts

```diff
@@ -94,6 +96,22 @@ export class AttachmentUnitFormComponent implements OnChanges {
         this.isFileTooBig.set(this.file.size > MAX_FILE_SIZE);
     }
 
+    get tooltipText(): string | null {
+        // Both name and file are invalid
+        if (!this.fileInputTouched && this.nameControl?.invalid) {
+            return this.translateService.instant('artemisApp.attachmentUnit.createAttachmentUnit.nameAndFileRequiredValidationError');
+        }
+        // Only file is invalid
+        if (!this.fileInputTouched) {
+            return this.translateService.instant('artemisApp.attachmentUnit.createAttachmentUnit.fileRequiredValidationError');
+        }
+        // Only name is invalid
+        if (this.nameControl?.invalid) {
+            return this.translateService.instant('artemisApp.attachmentUnit.createAttachmentUnit.nameRequiredValidationError');
+        }
+        return null;
+    }
```

Comment:
This approach will have a bad performance as the getter will be updated with ech cycle - you can add a console.log in there to see that you will have thousands of logs just by moving your mouse within the view.

While working on another PR I am introducing signals for the validation, you might want to stack this PR as you can then re-use the signals and instead of a getter use a computed property which is much more efficient regarding the performance.

(I will link the other PR once I have pushed the branch to remote and created it)

Reply:
You might also want to revise the client guidelines https://docs.artemis.cit.tum.de/dev/guidelines/client#null-and-undefined

Reply:
Thank you so much for the detailed feedback! I really appreciate it.
I'll wait until your PR is merged and then revise the code here.
I will also make sure I follow the guidelines 👍 
I guess `null` and `undefined` might be mistakes often made by new contributors. Is there perhaps a Coderabbit policy that could be added so that it warns when used?

Reply:
I rebased onto your PR, but I am not sure how I can use it. I have never worked with signals/computed values before, so I would be very happy if I could get some advice on this. 
I tried to rewrite the getter, but since it is only calculated when a signal changes, it was not always correct. That's why I added the isFormValid check at the beginning.
```js
tooltipText = computed(() => {
        // to trigger the computation
        if (this.isFormValid()) {
            return undefined;
        }

        if (!this.fileName() && this.nameControl?.invalid) {
            return this.translateService.instant('artemisApp.attachmentUnit.createAttachmentUnit.nameAndFileRequiredValidationError');
        }
        if (!this.fileName()) {
            return this.translateService.instant('artemisApp.attachmentUnit.createAttachmentUnit.fileRequiredValidationError');
        }
        if (this.nameControl?.invalid) {
            return this.translateService.instant('artemisApp.attachmentUnit.createAttachmentUnit.nameRequiredValidationError');
        }

        return undefined;
    });
```

Reply:
If you're new to signals, you could check out https://blog.angular-university.io/angular-signals/ or similar resources, to learn more about it.

I think your general approach here is correct, you just need to turn the things that change into signals

Reply:
Thanks for the advice!
I've added a signal for the name and restructured the tooltipText().
Would be great if you could take another look at this ;D